cmake/Doc.cmake

```diff
@@ -0,0 +1,16 @@
+function (run_doxygen)
+  find_package(Doxygen REQUIRED)
+
+  if (NOT DOXYGEN_DOT_FOUND)
+    message(FATAL_ERROR "Command `dot` not found.  Please install graphviz.")
```

Comment:
"Please install graphviz." Should this be doxygen?

Reply:
@RAMitchell No.  It's the `dot` command used to generate class hierarchy.